@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -306,11 +306,22 @@ const tmpCollisionBox = new CollisionBox();
 const tmpScreenPosition = new THREE.Vector2();
 const tmpTextOffset = new THREE.Vector2();
 const tmp2DBox = new Math2D.Box();
+const tmpCenter = new THREE.Vector2();
+const tmpSize = new THREE.Vector2();
 
 /**
  * The margin applied to the text bounds of every point label.
  */
 export const persistentPointLabelTextMargin = new THREE.Vector2(2, 2);
+/**
+ * Additional scaling (margin described as percentage of full size) applied to the new labels.
+ *
+ * This margin allows to account for imprecisions and changes in labels' bounds calculus when
+ * labels are changing their screen positions while camera is moving in the same plane.
+ * Those imprecisions may cause the label to appear or disappear when their bounds are in close
+ * proximity duo to collisions.
```

Comment:
That's only one use of the margin. In general the margin is there for any kind of camera movement. We want only to place new labels if they have a margin around them, so that slight camera changes in position/orientation doesn't case a  collision with neighboring labels.

Reply:
Would you like to change the comments then?

Reply:
yes better to give the whole picture, it doesn't mean to make the comment larger though. just mention a larger margin is used for new labels to ensure they'll stay on screen after small camera changes to avoid flickering.

Reply:
Ok, fixed it. Hope will be better now.